### PR TITLE
Upgrade akka 2.4.17, http 10.0.4, cassandra-query 0.24

### PR DESCRIPTION
**Pull request by jypma**
Also move to EventsByTagQuery2, since (1) is deprecated, and akka 2.5 will
promote v2 to be the actual EventsByTagQuery.

### CODE REVIEW

**Comment by jypma** [comment]
@vrbrahme @alar17 ready to review

File: ts-reaktive-actors/src/main/java/com/tradeshift/reaktive/akka/rest/EventRoute.java [hunk 1]
```diff
@@ -33,11 +38,11 @@
  */
 public class EventRoute {
     private static final MediaType.Binary mediaType =
-        MediaTypes.customBinary("application", "protobuf", true, 
-            HashMap.of("delimited", "true").put("messageType", "Query.EventEnvelope").toJavaMap(), 
-            true); 
+        MediaTypes.customBinary("application", "protobuf", true,
+            HashMap.of("delimited", "true").put("messageType", "Query.EventEnvelope").toJavaMap(),
```
Comment (vrbrahme) [inline ts-reaktive-actors/src/main/java/com/tradeshift/reaktive/akka/rest/EventRoute.java]: Seems like just a formatting change. Was this intended?

File: ts-reaktive-actors/src/main/java/com/tradeshift/reaktive/protobuf/EventEnvelopeSerializer.java [hunk 2]
```diff
@@ -3,12 +3,13 @@
...
 import akka.actor.ActorSystem;
-import akka.persistence.query.EventEnvelope;
+import akka.persistence.query.EventEnvelope2;
+import akka.persistence.query.TimeBasedUUID;
 import akka.serialization.Serialization;
 import akka.serialization.SerializationExtension;
 
 /**
- * Serializes and deserializes instances of {@link akka.persistence.query.EventEnvelope}, 
+ * Serializes and deserializes instances of {@link akka.persistence.query.EventEnvelope},
```
Comment (vrbrahme) [inline ts-reaktive-actors/src/main/java/com/tradeshift/reaktive/protobuf/EventEnvelopeSerializer.java]: formatting change again?

**Comment by vrbrahme** [comment]
few minor comments. Otherwise 👍

File: ts-reaktive-actors/src/main/java/com/tradeshift/reaktive/akka/rest/EventRoute.java (hunk 1 above)
Comment (jypma) [inline ts-reaktive-actors/src/main/java/com/tradeshift/reaktive/akka/rest/EventRoute.java]: yeah, my editor apparently removes trailing white space on save now. I'll move those lines out.

File: ts-reaktive-actors/src/main/java/com/tradeshift/reaktive/akka/UUIDs.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+     * since midnight, January 1, 1970 UTC. In particular, the timestamps
+     * returned by this method are comparable to the timestamps returned by
+     * {@link System#currentTimeMillis}, {@link java.util.Date#getTime}, etc.
+     *
+     * @param uuid the UUID to return the timestamp of.
+     * @return the Unix timestamp of {@code uuid}.
+     * @throws IllegalArgumentException if {@code uuid} is not a version 1 UUID.
+     */
+    public static long unixTimestamp(UUID uuid) {
+        if (uuid.version() != 1)
```
Comment (alar17) [inline ts-reaktive-actors/src/main/java/com/tradeshift/reaktive/akka/UUIDs.java]: Please keep the brackets for consistency.

**Comment by jypma** [comment]
moved whitespace changes to separate commit.

File: ts-reaktive-actors/src/main/java/com/tradeshift/reaktive/akka/UUIDs.java (hunk 3 above)
Comment (jypma) [inline ts-reaktive-actors/src/main/java/com/tradeshift/reaktive/akka/UUIDs.java]: this was a literal copy from cassandra...

**Pull request merged**